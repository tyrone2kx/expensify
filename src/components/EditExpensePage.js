import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { editExpense, removeExpense } from '../actions/expenses';


export class EditExpensePage extends React.Component {
    onSubmit = (expense) => {
        this.props.editExpense(this.props.expense.id, expense);    
        this.props.history.push('/');
    };

    onRemove = () => {
        this.props.removeExpense({ id: this.props.expense.id });
        this.props.history.push('/');
    }

    render() {
        return (
            <div>
                <ExpenseForm 
                expense={this.props.expense} 
                onSubmit={this.onSubmit} 
                />

                <button onClick={this.onRemove}> Remove </button>

            </div>
        );
    }
}


// this function defines what we want to get back from the store
// the props in this function was passed down from the HOC which is the default export below

const mapStateToProps = (state,props) => {
    return {
        expense: state.expenses.find((expense) => expense.id === props.match.params.id)
    }
};


const mapDispatchToProps = (dispatch, props) => ({
    editExpense: (id,expense) => dispatch(editExpense(id,expense)),
    removeExpense: (data) => dispatch(removeExpense(data))
});



export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);