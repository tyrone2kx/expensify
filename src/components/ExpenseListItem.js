import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';



const ExpenseListItem = ({ id, description, amount, createdAt}) => (
    <div>
    <Link to={`/edit/${id}`}>
        <h3>{description}</h3>
        <p>
        {numeral(amount / 100).format('$0,0.00')} - 
        
        {moment(createdAt).format('MMMM Do, YYYY')}
        
        </p>
    </Link>

        
    </div>
)


export default ExpenseListItem;