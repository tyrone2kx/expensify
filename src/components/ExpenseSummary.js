import React from 'react';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import getExpenseTotal from '../selectors/expenses-total';
import numeral from 'numeral';


export const ExpenseSummary = ({ count, total }) => {

        return (
            <div>
                <h1>Viewing {count} {count > 1 ? 'expenses' : 'expense'} totaling {numeral(total / 100).format('$0,0.00')} </h1>
            </div>
        );
    
}


const mapStateToProps = (state) => {
    return {
        expenses: selectExpenses(state.expenses, state.filters),
        count: selectExpenses(state.expenses, state.filters).length,
        total: getExpenseTotal(selectExpenses(state.expenses, state.filters))
    }
}

export default connect(mapStateToProps)(ExpenseSummary);