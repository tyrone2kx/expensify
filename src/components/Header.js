import React from 'react';
import {NavLink} from 'react-router-dom';

const Header = () => (
    <header>
        <h1>Expensify App</h1>
        <NavLink to="/" activeClassName="is-active" exact={true}>Dashboard</NavLink>
        <NavLink to="/help" activeClassName="is-active">Help</NavLink>
        <NavLink to="/create" activeClassName="is-active">Add Expense</NavLink>
    </header>
);

export default Header;