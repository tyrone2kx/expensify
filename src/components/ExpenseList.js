import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import selectExpenses from '../selectors/expenses';


export const ExpenseList = (props) => (
    <div>
    {
        props.expenses.length === 0 ? (<p>No Expenses</p>) : (
            props.expenses.map((expense) => {
                return <ExpenseListItem key={expense.id} {...expense} />
             })
        )
    }
     
    </div>
);


// this function defines what we want to get back from the store

const mapStateToProps = (state) => {
    return {
        expenses: selectExpenses(state.expenses, state.filters)
    }
}


// This kinda acts like the HOC. It passes the result from above to the second arguement (Component)

export default connect(mapStateToProps)(ExpenseList);


 