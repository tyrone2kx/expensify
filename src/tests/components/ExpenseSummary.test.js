import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseSummary } from '../../components/ExpenseSummary';


test('should render expense summary correctly', () => {

    const wrapper = shallow(<ExpenseSummary />);
    expect(wrapper).toMatchSnapshot();

});