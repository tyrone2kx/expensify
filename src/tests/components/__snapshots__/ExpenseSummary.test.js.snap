// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render expense summary correctly 1`] = `
<div>
  <h1>
    Viewing 
     
    expense
     totaling 
    $0.00
     
  </h1>
</div>
`;
